core/src/ucgi.c

```diff
@@ -0,0 +1,198 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include "game.h"
+#include "log.h"
+#include "sim.h"
+
+#define CMD_MAX 256
+
+#define MODE_SEARCHING 1
+#define MODE_STOPPED 0
+
+typedef struct GoParams {
+  int infinite;
+  int depth;
+  int stop_condition;
+  int threads;
+} GoParams;
+
+static Game *loaded_game = NULL;
+static Config *config = NULL;
+static Simmer *simmer = NULL;
+static int current_mode = MODE_STOPPED;
+
+int prefix(const char *pre, const char *str) {
+  return strncmp(pre, str, strlen(pre)) == 0;
+}
+
+void set_mode_stopped_callback() {
+  current_mode = MODE_STOPPED;
+  log_debug("setting current mode to stopped");
+}
+
+void load_position(const char *cgp, char *lexicon_name, char *ldname) {
+  char dist[50];
+  sprintf(dist, "data/letterdistributions/%s.csv", ldname);
+  char leaves[50] = "data/lexica/english.klv2";
+  char winpct[50] = "data/strategy/default_english/winpct.csv";
```

Comment:
Will the leaves and win tables be generalized to nonenglish in the future?

Reply:
yes, we can do that in the future. In the `if` statement after these statements, we can overwrite these variables depending on what the lexicon is.